controllers/consoleplugin.go

```diff
@@ -0,0 +1,405 @@
+package controllers
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+
+	"context"
+
+	consolepluginv1 "github.com/openshift/api/console/v1alpha1"
+	pipelinesv1alpha1 "github.com/redhat-developer/gitops-operator/api/v1alpha1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	resourcev1 "k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	gitopsPluginName             = "gitops-plugin"
+	displayName                  = "GitOps Plugin"
+	gitopsPluginSvcName          = gitopsPluginName + "-service"
+	proxyAlias                   = "gitops"
+	pluginImageEnv               = "GITOPS_CONSOLE_PLUGIN_IMAGE"
+	servicePort                  = 9001
+	pluginServingCertName        = "console-serving-cert"
+	kubeAppLabelApp              = "app"
+	kubeAppLabelComponent        = "app.kubernetes.io/component"
+	kubeAppLabelInstance         = "app.kubernetes.io/instance"
+	kubeAppLabelPartOf           = "app.kubernetes.io/part-of"
+	kubeAppLabelRuntimeNamespace = "app.kubernetes.io/runtime-namespace"
+	httpdConfigMapName           = "httpd-cfg"
+)
+
+func pluginDeployment() *appsv1.Deployment {
+	consolePluginImage := os.Getenv(pluginImageEnv)
+	if consolePluginImage == "" {
+		consolePluginImage = "quay.io/redhat-developer/gitops-console-plugin"
+	}
+	podSpec := corev1.PodSpec{
+		Containers: []corev1.Container{
+			{
+				Name:            gitopsPluginName,
+				Image:           consolePluginImage,
+				ImagePullPolicy: corev1.PullAlways,
+				Ports: []corev1.ContainerPort{
+					{
+						Name:          "http",
+						Protocol:      corev1.ProtocolTCP,
+						ContainerPort: servicePort,
+					},
+				},
+				VolumeMounts: []corev1.VolumeMount{
+					{
+						Name:      pluginServingCertName,
+						ReadOnly:  true,
+						MountPath: "/etc/httpd-ssl/certs/tls.crt",
+						SubPath:   "tls.crt",
+					},
+					{
+						Name:      pluginServingCertName,
+						ReadOnly:  true,
+						MountPath: "/etc/httpd-ssl/private/tls.key",
+						SubPath:   "tls.key",
+					},
+					{
+						Name:      httpdConfigMapName,
+						ReadOnly:  true,
+						MountPath: "/etc/httpd-cfg/httpd.conf",
+						SubPath:   "httpd.conf",
+					},
+				},
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceMemory: resourcev1.MustParse("128Mi"),
+						corev1.ResourceCPU:    resourcev1.MustParse("250m"),
+					},
+					Limits: corev1.ResourceList{
+						corev1.ResourceMemory: resourcev1.MustParse("256Mi"),
+						corev1.ResourceCPU:    resourcev1.MustParse("500m"),
```

Comment:
How did we derive the resource requests and limits values ? Any research ?

Reply:
I copied these settings from kam.go. How should I decide the value?

Reply:
You can create a  resource quota of 20 CPU and 20 GB on openshift-gitops namespace
after that, all the pods in the namespace can be created only if thy have resource requests and limits in their specification, just try to keep the resource limits as low as posible and run trail and error

this just for your understanding.

As of this PR, the requests and limits seems to be low already. I am ok with them